torch/_meta_registrations.py

```diff
@@ -2934,205 +2935,137 @@ def meta_addbmm(self, batch1, batch2, *, beta=1, alpha=1):
-    )
+            scalar_overload = "Scalar" if overload.startswith("Scalar") else "Tensor"
 
+            scalar_op_packet = getattr(aten, op_name.replace("_foreach_", ""))
+            scalar_op = getattr(
+                scalar_op_packet,
+                scalar_overload,
+                getattr(scalar_op_packet, "default", None),
+            )
+            _add_op_to_registry(
+                meta_table,
+                op,
+                partial(
+                    _meta_foreach_out_of_place,
+                    _scalar_op=scalar_op,
+                    _nlists=nlists + extra_nlists,
+                ),
+            )
 
-@register_meta(
-    [
-        aten._foreach_lerp_.Scalar,
-    ]
-)
-def meta__foreach_lerp__scalar(self, other, scalar=1):
-    _check_foreach_binop_tensor_lists(self, other)
+            op_name_ = f"{op_name}_"
+            op_ = getattr(getattr(aten, op_name_), overload)
+            scalar_op__packet = getattr(aten, op_name_.replace("_foreach_", ""), None)
+            if not scalar_op__packet:
+                continue
+            scalar_op_ = getattr(
+                scalar_op__packet,
+                scalar_overload,
+                getattr(scalar_op__packet, "default", None),
+            )
+            _add_op_to_registry(
+                meta_table,
+                op_,
+                partial(
+                    _meta_foreach_inplace,
+                    _scalar_op=scalar_op_,
+                    _nlists=nlists + extra_nlists,
+                ),
+            )
 
 
-@register_meta(
+register_meta_foreach(
     [
-        aten._foreach_addcdiv.Scalar,
-        aten._foreach_addcmul.Scalar,
-    ]
+        ("_foreach_abs", 1),
+        ("_foreach_acos", 1),
+        ("_foreach_asin", 1),
+        ("_foreach_atan", 1),
+        ("_foreach_ceil", 1),
+        ("_foreach_cos", 1),
+        ("_foreach_cosh", 1),
+        ("_foreach_erf", 1),
+        ("_foreach_erfc", 1),
+        ("_foreach_exp", 1),
+        ("_foreach_expm1", 1),
+        ("_foreach_frac", 1),
+        ("_foreach_floor", 1),
+        ("_foreach_lgamma", 1),
+        ("_foreach_log", 1),
+        ("_foreach_log10", 1),
+        ("_foreach_log1p", 1),
+        ("_foreach_log2", 1),
+        ("_foreach_neg", 1),
+        ("_foreach_reciprocal", 1),
+        ("_foreach_round", 1),
+        ("_foreach_sigmoid", 1),
+        ("_foreach_sign", 1),
+        ("_foreach_sin", 1),
+        ("_foreach_sinh", 1),
+        ("_foreach_sqrt", 1),
+        ("_foreach_tan", 1),
+        ("_foreach_tanh", 1),
+        ("_foreach_trunc", 1),
+        ("_foreach_zero", 1),
+        ("_foreach_add", 1),
+        ("_foreach_sub", 1),
+        ("_foreach_mul", 1),
+        ("_foreach_div", 1),
+        ("_foreach_maximum", 1),
+        ("_foreach_minimum", 1),
+        ("_foreach_clamp_min", 1),
+        ("_foreach_clamp_max", 1),
+        ("_foreach_addcdiv", 2),
+        ("_foreach_addcmul", 2),
+        ("_foreach_lerp", 2),
```

Comment:
Is it not possible to see that these should have a 2 rather than a 1 from the overloads directly?